videostore/renamer.go

```diff
@@ -0,0 +1,148 @@
+package videostore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+// renamer watches a directory and converts local timestamps to unix timestamps
+type renamer struct {
+	watchDir  string
+	outputDir string
+	logger    logging.Logger
+}
+
+func newRenamer(watchDir, outputDir string, logger logging.Logger) *renamer {
+	return &renamer{
+		watchDir:  watchDir,
+		outputDir: outputDir,
+		logger:    logger,
+	}
+}
+
+// processSegments periodically scans the directory for new files to process
+//
+// This function polls the directory regularly, looking for MP4 files
+// and processing them in order of creation.
+func (r *renamer) processSegments(ctx context.Context) error {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	r.logger.Debugf("Starting to scan directory: %s", r.watchDir)
+	for {
+		select {
+		case <-ctx.Done():
+			r.logger.Debug("Context done, stopping scanner and flushing remaining files")
+			if err := r.close(); err != nil {
+				return fmt.Errorf("error during close: %w", err)
+			}
+			return nil
+		case <-ticker.C:
+			if err := r.scanAndProcessFiles(); err != nil {
+				r.logger.Errorf("Error scanning directory: %v", err)
+			}
+		}
+	}
+}
+
+// scanAndProcessFiles scans the watch directory for MP4 files and processes them
+//
+// It excludes the most recent file to avoid processing it while it's still
+// being written by the segmenter.
+func (r *renamer) scanAndProcessFiles() error {
+	files, err := r.getSortedMPEGFiles()
+	if err != nil {
+		return err
+	}
+	if len(files) <= 1 {
+		r.logger.Debug("one or fewer active files found, skipping processing")
+		return nil
+	}
+
+	return r.processFiles(files[:len(files)-1]) // Exclude the most recent file
+}
+
+// getSortedMPEGFiles retrieves and sorts MP4 files in the watch directory
+func (r *renamer) getSortedMPEGFiles() ([]string, error) {
+	entries, err := os.ReadDir(r.watchDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read directory: %w", err)
+	}
+	var mpegFiles []string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			r.logger.Debugf("Skipping directory entry: %s", entry.Name())
+			continue
+		}
+		name := entry.Name()
+		if strings.HasSuffix(name, ".mp4") {
+			fullPath := filepath.Join(r.watchDir, name)
+			mpegFiles = append(mpegFiles, fullPath)
+		}
+	}
+	// Sort files by modification time (oldest first)
+	sort.Slice(mpegFiles, func(i, j int) bool {
+		infoI, _ := os.Stat(mpegFiles[i])
+		infoJ, _ := os.Stat(mpegFiles[j])
+		return infoI.ModTime().Before(infoJ.ModTime())
+	})
```

Comment:
1. Let's parse the filenames into time.TIme (we should already have a function to do that) and sort based on that, not the modification time.
2. Please only return files which are complete (aka for which you can parse out the duration)